libraries/freertos_plus/standard/freertos_plus_tcp/source/portable/NetworkInterface/Zynq/x_emacpsif_dma.c

```diff
@@ -422,8 +433,12 @@ int head = xemacpsif->rxHead;
 					addr |= XEMACPS_RXBUF_WRAP_MASK;
 				}
 				/* Clearing 'XEMACPS_RXBUF_NEW_MASK'       0x00000001 *< Used bit.. */
-				xemacpsif->rxSegments[ head ].address = addr;
 				xemacpsif->rxSegments[ head ].flags = 0;
+				xemacpsif->rxSegments[ head ].address = addr;
+				if (xemacpsif->rxSegments[ head ].address)
+				{
+					// Just to read it
```

Comment:
What is the meaning of this comment?

Reply:
I changed it to: 

~~~c
if (xemacpsif->rxSegments[ head ].address)
{
    /* Make sure that the value has reached the peripheral by reading it back. */
}

It works like a memory barrier.
~~~